segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/ContainerTableExtensionImpl.java

```diff
@@ -53,6 +54,7 @@
 import lombok.extern.slf4j.Slf4j;
 import lombok.val;
 
+import static io.pravega.shared.protocol.netty.WireCommands.NULL_TABLE_SEGMENT_OFFSET;
```

Comment:
No. Do not import this value. This code should not depend on `io.pravega.shared.protocol`.

Define an internal constant `NO_OFFSET`, set it to `-1`, and use it just in this class. Then in `commit`, invoke the appropriate API of `DirectSegmentAccess` based on whether you have this value or not. (actually you already make this check, so just change it to the internal value)